contracts/FeralfileArtworkV4.sol

```diff
@@ -10,60 +10,216 @@ import "./Authorizable.sol";
+        return _burnable;
+    }
+
+    function setBurnable(bool burnable_) external virtual onlyAuthorized {
+        _burnable = burnable_;
+    }
+
+    function bridgeable() public view virtual returns (bool) {
+        return _bridgeable;
+    }
+
+    function setBridgeable(bool bridgeable_) external virtual onlyAuthorized {
+        _bridgeable = bridgeable_;
+    }
+
+    function signer() public view virtual returns (address) {
+        return _signer;
+    }
+
+    function totalSupply() public view virtual returns (uint256) {
+        return _totalSupply;
+    }
+
+    function seriesMaxSupply(
+        uint256 seriesId
+    ) public view virtual returns (uint256) {
+        return _seriesMaxSupplies[seriesId];
+    }
+
+    function seriesTotalSupply(
+        uint256 seriesId
+    ) public view virtual returns (uint256) {
+        return _seriesTotalSupplies[seriesId];
+    }
 
-    /// @notice Update the IPFS cid of an artwork to a new value
-    function updateArtworkIPFSCid(
+    function getArtwork(
+        uint256 tokenId
+    ) public view virtual returns (Artwork memory) {
+        require(_exists(tokenId), "ERC721: token doesn't exist");
+        return _allArtworks[tokenId];
+    }
+
+    function setApprovalForAll(
+        address operator,
+        bool approved
+    ) public override(ERC721) onlyAllowedOperatorApproval(operator) {
+        super.setApprovalForAll(operator, approved);
+    }
+
+    function approve(
+        address operator,
+        uint256 tokenId
+    ) public override(ERC721) onlyAllowedOperatorApproval(operator) {
+        super.approve(operator, tokenId);
+    }
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public override(ERC721) onlyAllowedOperator(from) {
+        super.transferFrom(from, to, tokenId);
+    }
+
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId
+    ) public override(ERC721) onlyAllowedOperator(from) {
+        super.safeTransferFrom(from, to, tokenId);
+    }
+
+    function safeTransferFrom(
+        address from,
+        address to,
         uint256 tokenId,
-        string memory ipfsCID
-    ) external onlyAuthorized {}
+        bytes memory data
+    ) public override(ERC721) onlyAllowedOperator(from) {
+        super.safeTransferFrom(from, to, tokenId, data);
+    }
 
     /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
     function tokenURI(
         uint256 tokenId
     ) public view virtual override returns (string memory) {
-        return super.tokenURI(tokenId);
+        require(
+            _exists(tokenId),
+            "ERC721Metadata: URI query for nonexistent token"
+        );
+
+        string memory baseURI = _tokenBaseURI;
+        if (bytes(baseURI).length == 0) {
+            baseURI = "ipfs://";
+        }
+
+        return string(abi.encodePacked(baseURI, "/", tokenId.toString()));
```

Comment:
if the `baseURI == ""` then your return will be `ipfs://tokenID` which's not correct. we should return empty when we find the `baseURI == ""`, that mean the metadata is not ready yet